test/rgb_matrix/animation/config_test.exs

```diff
@@ -0,0 +1,30 @@
+defmodule ConfigTest do
+  use ExUnit.Case
+
+  describe "Animation configurations" do
+    test "can be created" do
+      assert %RGBMatrix.Animation.HueWave.Config{direction: :right, speed: 4, width: 20} ==
+               RGBMatrix.Animation.Config.new_config(
+                 RGBMatrix.Animation.HueWave.Config,
+                 RGBMatrix.Animation.HueWave.Config.schema(),
+                 %{}
+               )
```

Comment:
:warning: (recommended)

These config tests seem tightly coupled to the `HueWave` animation. Is there a way the generic config functionality can be tested here?

Here's some hasty pseudo code as an example of what I'm thinking:

```ex
test "can be created" do
  defmodule MockConfig do
    use RGBMatrix.Animation.Config

    field :speed, :integer
  end

  config =
    RGBMatrix.Animation.Config.new_config(
      MockConfig,
      MockConfig.schema(),
      %{speed: 5}
    )

  expected_config = %Config{
    module: MockConfig,
    schema: %{
      speed: %{type: :integer}
    },
    fields: %{
      speed: %{value: 5}
    }
  }

  assert config == expected_config
end
```

Reply:
Awesome, I'll do this. Thanks!

I'm not particularly fluent with Elixir tests, yet, so we should work on this more, some time.

Reply:
:+1: Sure! I'm happy to help out with this.